source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -1994,6 +1994,66 @@ SwiftLanguageRuntime::DoArchetypeBindingForType(StackFrame &stack_frame,
     // all conformances are always visible
     auto *module_decl = scratch_ctx->GetASTContext()->getStdlibModule();
 
+    // Replace opaque types with their underlying types when possible.
+    swift::Mangle::ASTMangler mangler(true);
+
+    while (target_swift_type->hasOpaqueArchetype()) {
+      auto old_type = target_swift_type;
+      target_swift_type = target_swift_type.subst(
+        [&](swift::SubstitutableType *type) -> swift::Type {
+          auto opaque_type =
+                          llvm::dyn_cast<swift::OpaqueTypeArchetypeType>(type);
+          if (!opaque_type)
+            return type;
+          
+          // Try to find the symbol for the opaque type descriptor in the
+          // process.
+          auto mangled_name = ConstString(
+                    mangler.mangleOpaqueTypeDescriptor(opaque_type->getDecl()));
+          
+          SymbolContextList found;
+          target.GetImages().FindSymbolsWithNameAndType(mangled_name,
+                                                        eSymbolTypeData, found);
+          
+          if (found.GetSize() != 1)
```

Comment:
What if you have two symbols from two different `dylibs` which resolve with the same name or type? At that point, shouldn't you probably just check that their value agree and pick one instead of returning early?